cpp/src/arrow/compute/exec/aggregate_node.cc

```diff
@@ -521,7 +538,7 @@ class GroupByNode : public ExecNode {
     if (ErrorIfNotOk(Consume(ExecSpan(batch)))) return;
 
     if (input_counter_.Increment()) {
-      ErrorIfNotOk(OutputResult());
```

Comment:
So we don't want to propagate the error back anymore? (I suppose that's all going away anyways?)

Reply:
Yes, at the moment, the only thing that handles this error is the sink node and all it will do is trigger an abort which will end up coming back here and marking this finished anyways.

That used to be needed because it was the only way to trigger an abort on a failure.

Now, with the async scheduler, any failed task will trigger an abort.  Furthemore, since `finished_` is a "task" it will trigger the abort in this path.